src/libs/TransactionUtils/index.ts

```diff
@@ -239,6 +240,11 @@ function getUpdatedTransaction(transaction: Transaction, transactionChanges: Tra
         shouldStopSmartscan = true;
     }
 
+    if (Object.hasOwn(transactionChanges, 'customUnitRateID')) {
+        updatedTransaction.modifiedCustomUnitRateID = transactionChanges.customUnitRateID;
```

Comment:
Ok thanks for explaining. Since it's only for optimistic data and we do something similar for modifiedWaypoints I'm  happy with it. I didn't know about modifiedWaypoints, thanks for explaining.